src/templates/components/CapiHostedCard.svelte

```diff
@@ -0,0 +1,131 @@
+<script context="module" lang="ts">
+	import { clickMacro } from '$lib/gam';
+</script>
+
+<script lang="ts">
+	import AudioIcon from './icons/AudioIcon.svelte';
+	import CameraIcon from './icons/CameraIcon.svelte';
+	import VideoIcon from './icons/VideoIcon.svelte';
+
+	export let card;
+
+	const { headline, image, url, audioTag, galleryTag, videoTag } = card;
+
+	const pictureSupported =
+		image?.sources?.length > 0 && 'srcset' in new Image();
+</script>
+
+<a href={clickMacro(url)}>
+	<div class="media">
+		{#if pictureSupported}
+			<picture>
+				{#each image.sources.filter((source) => source.minWidth === '0') as source}
```

Comment:
just wondering where `image.sources` gets filled, I can only see it get's set to an empty array [here](https://github.com/guardian/commercial-templates/pull/348/files#diff-ac7847cc5681b3003118d86d922e6e84cc33349dbb41ceacc66d495184167c66R36)

Reply:
`image.sources` comes from the Capi api. It gets overridden here if the user specifies a different image that should used in place of the api. If no image is specified in the GAM overrides, then `image.sources` won't be replaced.

Reply:
Looking at the code https://github.com/guardian/commercial-templates/pull/348/files#diff-ac7847cc5681b3003118d86d922e6e84cc33349dbb41ceacc66d495184167c66R38-R40

if a gam override is specified `image.sources` is an empty array, if it isn't image.sources will be undefined, unless I'm missing something?

Reply:
If a GAM override is not specified, `image` will be equal to `cardData[i]?.articleImage`. `articleImage` has `sources` and `backupSrc` on it. There's an example of what `articleImage` looks like here: https://api.nextgen.guardianapps.co.uk/commercial/api/capi-multiple.json?k=advertiser-content%2Fipo-awareness

Reply:
Ahhhh, sorry, my band, my brain couldn't fathom that `articleImage` could be an object!🙈